### PR TITLE
Support compound conditions in when clauses

### LINKED ISSUE #124: Case tag does not honor compound conditions in when clauses
This display the correct season

```
        {% assign m = post.date | date: "%b" %}
        {% case m %}
          {% when 'April' %}Spring
          {% when 'May'  %}Spring
          {% when 'June' %}Spring
          {% when 'July' %}Summer
          {% when 'August' %}Summer
          {% when 'September' %}Summer
          {% when 'October' %}Autumn
          {% when 'November' %}Autumn
          {% when 'December' %}Autumn
          {% when 'January'  %}Winter
          {% when 'February' %}Winter
          {% when 'March' %}Winter
          {% else %}{{ post.date | date: "%b" }}.
          {% endcase %}
        {{ post.date | date: "%Y" }}
```

Whereas this only display the full month name (the else clause)
```
        {% assign m = post.date | date: "%b" %}
        {% case m %}
          {% when 'April' or 'May' or 'June' %}Spring
          {% when 'July' or 'August' or 'September' %}Summer
          {% when 'October' or 'November' or 'December' %}Autumn
          {% when 'January' or 'February' or 'March'  %}Winter
          {% else %}{{ post.date | date: "%b" }}.
          {% endcase %}
        {{ post.date | date: "%Y" }}
```

This last option works in Jelyll.

### PR DESCRIPTION
**Pull request by live627**
Fix #124

### CODE REVIEW

File: src/Liquid/Tag/TagCase.php [hunk 1]
```diff
@@ -99,14 +99,13 @@ public function endTag()
 	 */
 	public function unknownTag($tag, $params, array $tokens)
 	{
-		$whenSyntaxRegexp = new Regexp('/' . Liquid::get('QUOTED_FRAGMENT') . '/');
-
 		switch ($tag) {
 			case 'when':
+				$whenSyntax = preg_match_all('/(?<=,|or|^)\s*((["\'])?(?(2)(?:(?!\2).)*+\2|\b\w+))/', $params, $matches, PREG_UNMATCHED_AS_NULL);
```
Comment (sanmai) [inline src/Liquid/Tag/TagCase.php:104]: What is the reason we're not using `Liquid::get('QUOTED_FRAGMENT')` here? What if someone uses a different regex for `QUOTED_FRAGMENT`?

Comment (live627) [reply src/Liquid/Tag/TagCase.php]: I put that one back in.

**Comment by live627** [comment]
This one is ready to be merged in.

File: tests/Liquid/Tag/TagCaseTest.php [hunk 2]
```diff
@@ -49,6 +49,26 @@ public function testCase()
...
+		yield [[], '{% assign handle = "cookie" %}{% assign condition = "cookie" %}{% case handle %}{% when "cake" %}This is a cake{% when condition, "biscuit" %}This is a cookie{% else %}This is not a cake nor a cookie{% endcase %}', 'This is a cookie'];
+	}
+
+	/**
+	 * @dataProvider multipleConditionsProvider
+	 */
+	public function testMultipleConditions(array $assigns, string $test, string $expected)
+	{
+		$this->assertTemplateResult($expected, $test, $assigns);
+	}
```
Comment (sanmai) [inline tests/Liquid/Tag/TagCaseTest.php:70]: The existing implementation passes this battery of tests. How can we confirm there was an issue?

Please merge with `master` before making changes here.

Comment (live627) [reply tests/Liquid/Tag/TagCaseTest.php]: The confirmation is in a fixture that I had to update.

**Comment by sanmai** [comment]
Released in 1.4.36.

**Pull request merged**